enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,373 @@
+dev-scripts repository that run the agent flows. The purpose of this would be to find 
+breakages in the tests across the different teams.
+
+** The dev-scripts repository clones the [metal-dev-env](https://github.com/metal3-io/metal3-dev-env) 
+repository for most of its Ansible roles. Which means there is a risk of changes in this 
+repository can break things in our repository as well. We have less control over this repository 
+than others inside Openshift.
+
+This is partially mitigated already because the repository is cloned at a specific
+Git SHA. To further this we can add a test in dev-scripts that runs maybe once or twice
+a week that clones the HEAD of this repository. We would be alerted for breakages more
+frequently and we can mitigate them on a case-by-case basis.
+
+
+## Design Details
+
+### Identified Overlap Points for Agent Installer
+
+During the discussions with the Metal Platform team it was identified that the following
+components of dev-scripts are the likely points of overlap for the new agent installer project:
+
+* The [01_install_requirements.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/01_install_requirements.sh) 
+and [02_configure_host.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/02_configure_host.sh) 
+scripts that are run using `make`.
+* Ansible playbooks to set up VMs such as: [vm_setup_vars.yml](https://github.com/openshift-metal3/dev-scripts/blob/master/vm_setup_vars.yml)
+
+These will be the things that we will focus on first when refactoring parts of dev-scripts.
+
+### Overall Components of the e2e Framework
+
+Using the general flow of an e2e test we can sort the overall components of the e2e framework.
+A general flow of an e2e test has the following phases:
+
+* Pre Environment Setup: 
+** Setting up VMs, networks, locations of images, image registries, etc
+* Build: 
+** Building the agent-based installer ISO and container images associated with it
+* Test:
+**  Test scenarios with pass/failure outputs
+* Reporting:
+** Reports comprised of test results
+* Environment Teardown
+** Removing VMs, networks, etc to start fresh again
+
+#### Pre Environment Setup 
+
+* The e2e framework will be a collection of Ansible playbooks and roles
+* Using roles to define reusable steps scoped to units of work
+* Use playbooks to set up common e2e workflows and chains for the project
+
+In discussions regarding this enhancement we identified the dev-scripts has components that are
+a part of the Pre Environment Setup and Environment Teardown phases of the framework. We will
+move forward with adding and potentially refactoring scripts in this repository for these phases.
+
+#### Build
+
+The scripts/automation for building the agent-based installer should be inside the repository
+where the installer is being worked on and not dev-scripts. We will reference the built 
+container images and ISO in dev-scripts from those repositories. Currently this project is being 
```

Comment:
How can we substitute a container image with a local build for development?

Reply:
I removed this part entirely. In general the pattern I was describing is that the developer would build the container images they want to test themselves. We would probably have scripts and/or Makefile targets to make that easy. But the root of it is that you might have several branches and your want to build different containers on the fly for your changes. You want more control than a git clone allows. If this makes sense.